rclcpp/src/rclcpp/generic_publisher.cpp

```diff
@@ -31,4 +31,47 @@ void GenericPublisher::publish(const rclcpp::SerializedMessage & message)
   }
 }
 
+void GenericPublisher::publish_loaned_msg(const rclcpp::SerializedMessage & message)
+{
+  if (this->can_loan_messages()) {
+    auto loaned_message = borrow_loaned_message();
+    deserialize_message(message.get_rcl_serialized_message(), loaned_message);
+    publish_loaned_message(loaned_message);
+  } else {
+    throw std::runtime_error("failed to loan messages");
```

Comment:
more direct message would be suitable to tell use that loan message cannot be handled with current middleware?

```suggestion
    throw std::runtime_error("current middleware cannot handle loan messages");
```